screenpipe-vision/src/apple.rs

```diff
@@ -1,86 +1,135 @@
 use image::DynamicImage;
 use log::error;
-use std::ffi::CStr;
-use std::os::raw::{c_char, c_uchar};
+use serde::{Deserialize, Serialize};
+use std::path::PathBuf;
+use uuid::Uuid;
 
-use std::ops::Drop;
-
-struct OcrResultGuard(*mut c_char);
+#[derive(Serialize, Deserialize, Debug)]
+struct OcrResultBBox {
+    x: f64,
+    y: f64,
+    height: f64,
+    width: f64,
+}
 
-impl Drop for OcrResultGuard {
-    fn drop(&mut self) {
-        unsafe {
-            if !self.0.is_null() {
-                free_string(self.0);
-            }
-        }
-    }
+#[derive(Serialize, Deserialize, Debug)]
+struct OcrTextElement {
+    bounding_box: Vec<OcrResultBBox>,
+    confidence: f32,
+    text: String,
 }
 
-#[cfg(target_os = "macos")]
-#[link(name = "screenpipe")]
-extern "C" {
-    fn perform_ocr(
-        image_data: *const c_uchar,
-        length: usize,
-        width: i32,
-        height: i32,
-        languages: *const *const c_char,
-        languages_count: i32,
-    ) -> *mut c_char;
-    fn free_string(ptr: *mut c_char);
+#[derive(Serialize, Deserialize, Debug)]
+struct OcrResult {
+    ocr_result: String,
+    text_elements: Vec<OcrTextElement>,
+    overall_confidence: f32,
 }
+
 #[cfg(target_os = "macos")]
 pub fn perform_ocr_apple(
     image: &DynamicImage,
-    languages: Vec<screenpipe_core::Language>,
+    _languages: Vec<screenpipe_core::Language>,
```

Comment:
we need the language feature

Reply:
Cidre doesn't allow langauge selection. That's why I had used revision.

Reply:
Will be adding languages support by tmrw (lost the code when switching branches).